.github/workflows/cypress.yml

```diff
@@ -119,22 +119,31 @@ jobs:
           node-version: '14'
           cache: npm
 
-      - name: Cache dependencies
-        id: cache
-        uses: actions/cache@v3
-        with:
-          path: ./node_modules
-          key: node_modules-${{ hashFiles('package-lock.json') }}
-
       - name: upgrade npm
         run: |
           npm install --global npm@8.5
+              # Restore the previous NPM modules and Cypress binary archives.
+
+      # source: https://github.com/cypress-io/cypress-example-kitchensink/blob/master/.github/workflows/single.yml
```

Comment:
This source is quite old, a lot has changed since then :p 
using `cache: npm` on the `setup/node` step will already cache the `npm cache`.